library/x509_crt.c

```diff
@@ -3005,6 +3005,127 @@ static int x509_crt_verify_chain(
     }
 }
 
+#ifdef _WIN32
+/* ??? */
+#elif defined(__sun)
+/* Solaris requires -lsocket -lnsl for inet_pton() */
+#elif defined(__has_include)
+#if __has_include(<sys/socket.h>)
+#include <sys/socket.h>
+#endif
+#if __has_include(<arpa/inet.h>)
+#include <arpa/inet.h>
+#endif
+#else
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#endif
+
+#ifndef AF_INET6
```

Comment:
Please add a note here to explain that “is `AF_INET6` defined?” stands for “we're confident that this platform has an `inet_pton` function that supports both IPv4 and IPv6”.

Reply:
```
/* Use whether or not AF_INET6 is defined to indicate whether or not to use
 * the platform inet_pton() or a local implementation (below).  The local
 * implementation may be used even in cases where the platform provides
 * inet_pton(), e.g. when there are different includes required and/or the
 * platform implementation requires dependencies on additional libraries.
 * Specifically, Windows requires custom includes and additional link
 * dependencies, and Solaris requires additional link dependencies.
 * Also, as a coarse heuristic, use the local implementation if the compiler
 * does not support __has_include(), or if the definition of AF_INET6 is not
 * provided by headers included (or not) via __has_include() above. */
```
:rocket: